openhands/runtime/plugins/agent_skills/utils/aider/linter.py

```diff
@@ -109,6 +111,55 @@ def py_lint(self, fname, rel_fname, code):
             error = basic_lint(rel_fname, code)
         return error
 
+    def ts_lint(self, fname, rel_fname, code):
+        """Use typescript compiler to check for errors."""
+        tsc_cmd = 'tsc --noEmit --allowJs --checkJs --strict --noImplicitAny --strictNullChecks --strictFunctionTypes --strictBindCallApply --strictPropertyInitialization --noImplicitThis --alwaysStrict'
+        try:
+            tsc_res = self.run_cmd(tsc_cmd, rel_fname, code)
+            if tsc_res:
+                # Parse the TSC output
+                error_lines = []
+                for line in tsc_res.text.split('\n'):
+                    if ': error TS' in line or ': warning TS' in line:
+                        parts = line.split(':')
+                        if len(parts) >= 3:
+                            try:
+                                error_lines.append(int(parts[1]))
+                            except ValueError:
+                                continue
+                return LintResult(text=tsc_res.text, lines=error_lines)
+        except FileNotFoundError:
+            pass
+
+        # If tsc is not available or finds no errors, fall back to basic linting
+        basic_lint_result = basic_lint(rel_fname, code)
+        if basic_lint_result:
+            # Convert basic lint result to TypeScript-like error message
+            ts_error_message = f'{rel_fname}({basic_lint_result.lines[0]},1): error TS2304: {basic_lint_result.text}'
```

Comment:
Can we unify the linting result format? https://github.com/All-Hands-AI/OpenHands/blob/7cc0b2827a70ac12fc6d5b697ba616be3873d31c/openhands/runtime/plugins/agent_skills/utils/aider/linter.py#L157-L169 has assumptions on error message format

Reply:
FWIW, I am requesting because I am working on https://github.com/OpenDevin/OpenDevin/issues/3412, which needs line numbers.

Reply:
This is the unification, so to speak. E.g. the typescript linting method (ts_lint) tries to return the same format, since tsc' format is different like `file.ext(3,1)`. Linters for other languages may be different, too.
If further changes are actually needed, we should do that in a followup PR.

Reply:
Can you add a TODO comment here?

Reply:
Sure